sys/include/net/ng_llhdr.h

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net
+ * @{
+ *
+ * @file
+ * @brief       Generic link layer header
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef LLHDR_H_
+#define LLHDR_H_
+
+#include <string.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Generic link layer header
+ *
+ * The actual addresses are put into the memory following this datastructure.
+ */
+typedef struct __attribute__((packed)) {
+    uint8_t addr_len;           /**< link layer address length in byte */
+    uint8_t rssi;               /**< rssi of received packet (optional) */
+    uint8_t lqi;                /**< lqi of received packet (optional) */
+    uint8_t options;            /**< any optional options */
+} llhdr_t;
```

Comment:
At least for multi-interface support in NHDP and OLSRv2 which are working atop of IPv6 it is necessary that the protocols can determine on which interface a packet was received on. I haven't looked into the whole new network stack structure yet, so I can't assess whether this is the right struct to save this information but generally the information has to be available somewhere for network layer protocols.